scripts/release/install.ps1

```diff
@@ -49,7 +49,7 @@ $ErrorActionPreference = "Stop"
 $version = "0.3.0"
 # endregion replace-version unstable
 
-$toolchain = "nightly-2023-08-24"
+$toolchain = "nightly-2023-10-05"
```

Comment:
Looking at this and the documentation update changes, I think, it would be better to switch the nightly toolchain script to also use region markers. This would also allow us to have different variations. This install script still says version v0.3.0, but now has a newer toolchain, which is not ideal...

Reply:
Vector uses unstable trait specialization feature, so it should definitely be optimized for the case when the `Vec` is extend with another `Vec` ([code](https://github.com/rust-lang/rust/blob/4f4a413fe6931d0ad9d3ac6bd20ff36398961e64/library/alloc/src/vec/mod.rs#L2823)).